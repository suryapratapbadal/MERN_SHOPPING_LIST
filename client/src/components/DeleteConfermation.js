import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { deleteItem } from '../actions/index';

export class DeleteConfermation extends React.Component {
    state = {
        open: this.props.toggleDeleteConfirmation,
    };

    componentDidUpdate(prevProps) {

        if (prevProps.toggleDeleteConfirmation !== this.props.toggleDeleteConfirmation) {
            this.setState({ open: this.props.toggleDeleteConfirmation });
        }
    }

    // handleClickOpen = () => {
    //     this.setState({ open: true });
    // };

    handleClose = () => {
        this.setState({ open: false });
    };

    handleCloseAgree = () => {
        this.props.deleteItem(this.props.id);
        this.setState({ open: false });
    }

    render() {
        return (
            <div>
                
                <Dialog
                    open={this.state.open}
                    onClose={this.handleClose}
                    aria-labelledby="alert-dialog-title"
                    aria-describedby="alert-dialog-description"
                >
                    <DialogTitle id="alert-dialog-title">confirm Delete</DialogTitle>
                    <DialogContent>
                        <DialogContentText id="alert-dialog-description">
                            Are you sure you want to delete this Item?
                        </DialogContentText>
                    </DialogContent>
                    <DialogActions>
                        <Button onClick={this.handleClose} color="primary">
                            Disagree
                        </Button>
                        <Button onClick={this.handleCloseAgree} color="primary" autoFocus>
                            Agree
                        </Button>
                    </DialogActions>
                </Dialog>
            </div>
        );
    }
}

export default connect(state => {
    const loading = state.itemReducer.loading || false;
    return {
        loading
    }
}, dispatch => {
    return bindActionCreators({ deleteItem: deleteItem }, dispatch)
}
)(DeleteConfermation);