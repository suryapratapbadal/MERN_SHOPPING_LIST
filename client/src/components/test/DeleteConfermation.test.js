import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import { DeleteConfermation } from '../DeleteConfermation';


const mockdeletefn = jest.fn();
const toggleDeleteConfirmationfn = jest.fn();
const id ='1';


// Component Renders
describe('<DeleteConfermation />', () => {

    const wrapper = shallow(<DeleteConfermation deleteItem={mockdeletefn} _id={id} toggleDeleteConfirmation={toggleDeleteConfirmationfn}/>);
    const event = {
        stopPropagation: jest.fn(),
        preventDefault: jest.fn(),
        target: { value : 'dummy'}
    }
    describe('render()', () => {

        test('renders the component', () => {

            expect(toJson(wrapper)).toMatchSnapshot();

        });
    });


    // describe('onChange()',()=>{
    //     test('handle click on button',()=>{
    //         wrapper.find('WithStyles(Checkbox)').props().onChange();
    //         expect(wrapper.state('selected')).toEqual(true);
    //     });
    // });

    // describe('onChange()',()=>{
    //     test('handle click on button',()=>{
    //         wrapper.find('WithStyles(TextField)').at(0).props().onChange(event);
    //         expect(wrapper.state('itemName')).toEqual('dummy');
    //     });
    // });

    describe('onClose()',()=>{
        test('handle click on button',()=>{
            wrapper.find('WithStyles(Dialog)').props().onClose();
            expect(wrapper.state('open')).toEqual(false);
        });
    });

    describe('onClick()',()=>{
        test('handle click on button',()=>{
            wrapper.find('WithStyles(Button)').at(1).props().onClick();
            expect(wrapper.state('open')).toEqual(false);
        });
    });
});