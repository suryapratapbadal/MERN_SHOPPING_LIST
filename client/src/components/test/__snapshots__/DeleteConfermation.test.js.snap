// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteConfermation /> render() renders the component 1`] = `
<div>
  <WithStyles(Dialog)
    aria-describedby="alert-dialog-description"
    aria-labelledby="alert-dialog-title"
    onClose={[Function]}
    open={[Function]}
  >
    <WithStyles(DialogTitle)
      id="alert-dialog-title"
    >
      confirm Delete
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)
        id="alert-dialog-description"
      >
        Are you sure you want to delete this Item?
      </WithStyles(DialogContentText)>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
      >
        Disagree
      </WithStyles(Button)>
      <WithStyles(Button)
        autoFocus={true}
        color="primary"
        onClick={[Function]}
      >
        Agree
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;
